### ISSUE TITLE
Unable to install Advanced Ads and other ad plugins

### ISSUE THREAD

**Issue by arunsathiya** [Automattic/wp-calypso#58322]
### Quick summary

Installing [Advanced Ads](https://wordpress.com/plugins/advanced-ads/) plugin fails on Calypso/marketplace.

### Steps to reproduce

- Start with an Atomic site.
- Visit https://wordpress.com/plugins/advanced-ads/
- Click on "Install and activate"

### What you expected to happen

The plugin installs fine.

### What actually happened

The installation fails:

<img width="1787" alt="image" src="https://user-images.githubusercontent.com/18581859/142720662-2509fa74-cf0f-4536-bfc9-27632c244a90.png">

Console shows this:

```
Uncaught (in promise) Error: 
    e https://wordpress.com/calypso/evergreen/70520.abec2986cc75eac32477.min.js:2
    k https://wordpress.com/calypso/evergreen/73548.013f6b8b14d7a4bdd3e8.min.js:2
```

### Context

User report: 32827212-hc.

### Operating System

macOS

### Browser

Mozilla Firefox

### Simple, Atomic or both?

Atomic

### Theme-specific issue?

No.

### Other notes

- Other plugins like Yoast, Mailpoet and All-in-one WP migration install fine. 
- Advanced Ads is compatible: https://wordpress.com/support/incompatible-plugins/

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

No and the platform is unusable

**Comment by arunsathiya** [comment]
Flagged internally too: p1637398849218200-slack-C029JEQRVRT

**Comment by gavande1** [comment]
I've tried to install an [Advanced Ads](https://wordpress.com/plugins/advanced-ads) plugin on the test site. It worked well as expected. Then I went ahead and tried to install the same plugin on the user's site. It got installed without an issue. Every time it seems to be working fine. 
(I might be missing something here.)

![](https://cloudup.com/cE0UTx9588K+)

**Comment by arunsathiya** [comment]
While discussing with @gavande1 on Slack, I found that "uBlock Origin" extension (a popular ad-blocker for browsers) was the cause of the issue. When it's active, this plugin is blocked. We suspect it's blocking only ad-type plugins.. because it's an ad-blocker. That's why installing other plugins like Yoast work fine.

Here's a video of what I see on Firefox:




https://user-images.githubusercontent.com/18581859/142755315-c9a62a6b-3719-4aae-8b7e-22de71c5ef80.mov

And on Edge:

https://user-images.githubusercontent.com/18581859/142755258-63a7eb90-4d92-4b84-965f-72536d95ed99.mov

I'll leave this open so that the product team can discuss how to proceed. Say, talk to uBlock Origin/Advanced Ads team to figure out a way forward?

**Comment by cpapazoglou** [comment]
Can we try using a `POST` rather than `GET` request to pass the `slug` parameter and workaround the ads blocker?

**Comment by cpapazoglou** [comment]
We will pick this issue in our sprint so that we research possible solutions.

**Comment by kriskorn** [comment]
@cpapazoglou Thank you for looking into this. 

Just wanted to note that we have another case here - p9F6qB-7HW-p2 and it was not possible to install Elementor Pro nor Hello Dolly on the user site. I was able to install the Hello Dolly plugin on my test site. So I guess it is not only ad blocking extensions, but maybe something else that may be blocking the installation of plugins via this flow on certain site setups.

**Comment by jartes** [comment]
Noting that we had another case yesterday on p1637680765469600-slack-C03TY6J1A, where the user was getting a `Upload problem: json_syntax` error: 

<img width="935" alt="Screen Shot 2021-11-23 at 10 23 09" src="https://user-images.githubusercontent.com/1046943/143232400-53fa515f-5943-4d2b-bbec-9d38b2328901.png">

I was able to reproduce on their site, but not in a test site I have using Cloudflare.

**Comment by cpapazoglou** [comment]
@kriskorn, @jartes, @arunsathiya 

Seems like we are having three different cases here:

### Adblocker case 
Plugins with `ads` terms in their slug get blocked by ad blockers. Also, any request containing `ads` terms in the URL will get blocked. For example the plugin icon won't load 
![SS 2021-11-24 at 19 07 08](https://user-images.githubusercontent.com/12430020/143283593-7331c557-3b9a-41c2-b419-e3fdc9101389.png)

We can chose one of the following "solutions":
- detect an add blocker and suggest to the user to disable it or visit `/wp-admin/plugin-install.php`, they will still lose the images though. Anecdotally, not all plugins with `ads` term are blocked.
- change the `install` endpoint to `POST` as @gavande1 suggested seems working for me. 
https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/packages/wpcom.js/src/lib/site.plugin.js#L74
and then `Jetpack_JSON_API_Plugins_Endpoint` in wpcom

### Cloudflare case
- a cloudflare chalenge blocking the user from uploading through Calypso. From WP Admin, you get the Cloudflare page which you can then bypass. @jartes can we confirm it's a cloudflare problem so we can then try to resolve it internally or escalate it to cloudflare?
<img width="1594" alt="SS 2021-11-24 at 17 47 13" src="https://user-images.githubusercontent.com/12430020/143274285-1999d925-908d-4e93-8798-923827c601bb.png">

@kriskorn, I couldn't grok the problem in p9F6qB-7HW-p2. It's neither of the above. Is there a chance it's something  Pressable related? Also, that user can't upload images too.

While we try to fix the above, please suggest to the users using `/wp-admin/plugin-install.php` for installing from wp.org repo or uploading a plugin.

**Comment by jartes** [comment]
@cpapazoglou We are still waiting user reply on 4539352-zd-woothemes to check if they have any firewall rule in place.

**Comment by vinnykaur** [comment]
Had one more in hc-32960053. Installing via .../wp-admin/plugin-install.php helped.

**Comment by cpapazoglou** [comment]
We have decided:

- change the `install` endpoint to `POST` as @gavande1 suggested seems working for me. 
https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/packages/wpcom.js/src/lib/site.plugin.js#L74
and then `Jetpack_JSON_API_Plugins_Endpoint` in wpcom

so that we solve the `ads` problem. The cloudflare problem will be resolved in a separate issue as soon as we have more feedback.

**Comment by cpapazoglou** [comment]
@JulianBustamante what is the status here? do you need any help?

**Comment by JulianBustamante** [comment]
@cpapazoglou I was working with @gavande1 and finished the local Jetpack setup. I'm still gathering information about how the WP.com and Jetpack endpoints are working to make the required changes.
